src/hooks/useAntsForWallet.tsx

```diff
@@ -0,0 +1,115 @@
+    useAccessControlList();
+  const { data: antVersions = {} } = useANTVersions();
+  const { data: latestAntVersion } = useLatestANTVersion();
+  return useQuery({
+    queryKey: ['ants-with-metadata', walletAddress?.toString()],
+    queryFn: async () => {
+      const antAo = connect(aoNetwork.ANT);
+      const uniqueAnts = [
+        ...new Set([
+          ...(accessControlList?.Owned ?? []),
+          ...(accessControlList?.Controlled ?? []),
+        ]),
+      ];
+      const antsWithMetadata = await Promise.all(
+        uniqueAnts.map(async (processId) => {
+          const [state, processMeta] = await Promise.all([
+            queryClient
+              .fetchQuery(
+                buildAntStateQuery({
+                  processId,
+                  ao: antAo,
+                  hyperbeamUrl: hyperbeamUrl,
+                }),
+              )
+              .catch((e) => {
+                console.error(e);
+                return undefined;
+              }),
+            // TODO: send these in a single request
+            queryClient
+              .fetchQuery(
+                buildGraphQLQuery(aoNetwork.ANT.GRAPHQL_URL, {
+                  ids: [processId],
+                }),
+              )
+              .then((res) => res?.transactions.edges[0].node)
+              .catch((e) => {
+                console.error(e);
+                return null;
+              }),
+          ]);
+          const moduleId = processMeta?.tags.find(
+            (tag) => tag.name === 'Module',
+          )?.value;
+
+          const version = Object.keys(antVersions ?? {}).find(
+            (versionNumber) =>
+              antVersions?.[versionNumber].moduleId === moduleId,
+          );
+
+          const isLatestVersion =
+            processMeta?.tags.find((tag) => tag.name === 'Module')?.value ===
+            latestAntVersion?.moduleId;
+
+          return {
+            [processId]: {
+              state,
+              version: version ? parseInt(version) : 0,
+              processMeta: processMeta as unknown as TransactionEdge['node'],
+              isLatestVersion,
+              errors: [],
+            },
+          };
+        }),
+      );
+
+      return antsWithMetadata.reduce((acc, item) => ({ ...acc, ...item }), {});
+    },
+  });
+};
+
+export const useAntsRequireUpdate = (): {
+  ants: string[];
+  isLoading: boolean;
+} => {
+  const { data: antData = {}, isLoading, isRefetching } = useAntsForWallet();
+  const {
+    data: latestAntVersion,
+    isLoading: isLoadingLatestAntVersion,
+    isRefetching: isRefetchingLatestAntVersion,
+  } = useLatestANTVersion();
+  return {
+    ants: Object.keys(antData)
+      .filter((processId) => {
+        return +antData[processId].version < +(latestAntVersion?.version || 0);
+      })
+      .map((processId) => processId),
+    isLoading:
+      isLoading ||
+      isLoadingLatestAntVersion ||
+      isRefetching ||
+      isRefetchingLatestAntVersion,
+  };
+};
```

Comment:
recommend useMemo on the ants and isLoading vars

deps for ants is the query data (antData and latestAntVersion) then deps for isLoading is all the loading variables.

This helps prevent unnecessary rerenders, especially if using this in a lot of elements (eg domains table on every single row and in ternaries for feature blocking inputs)